src/main/webapp/components/directives/cache/general/general.js

```diff
@@ -0,0 +1,37 @@
+(function () {
+  'use strict';
+
+  var module = angular.module('ispn.directives.cache.general', ['ispn.services.utils']);
+
+  module.directive('general', ['utils', function (utils) {
+      return {
+        restrict: 'E',
+        scope: {
+          data: '=',
+          initDefaults: '=',
+          readOnly:'='
+        },
+        replace: true,
+        templateUrl: 'components/directives/cache/general/general.html',
+        link: function (scope, element, attrs) {
+          scope.cacheTypes = ['distributed-cache', 'local-cache','replicated-cache', 'invalidation-cache'];
+          scope.cacheModes = [ 'SYNC', 'ASYNC'];
+
+          if (scope.initDefaults){
+            scope.data.type = 'distributed-cache';
+            scope.data.data = {};
```

Comment:
When creating new object, I think it is somewhat more intuitive/clearer to do it like this:
```
a = {
  b: 0
};
```
then like this
```
a = {};
a.b = 0;
```